letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -545,8 +574,15 @@ if [ "$1" = "--le-auto-phase2" ]; then
     # If the selected Bootstrap function isn't a noop and it differs from the
     # previously used version
     if [ -n "$BOOTSTRAP_VERSION" -a "$BOOTSTRAP_VERSION" != "$PREV_BOOTSTRAP_VERSION" ]; then
-      # if non-interactive mode or stdin and stdout are connected to a terminal
-      if [ \( "$NONINTERACTIVE" = 1 \) -o \( \( -t 0 \) -a \( -t 1 \) \) ]; then
+      # Check if we can rebootstrap without manual user intervention: this require that
+      # certbot-auto is in non-interactive mode AND selected bootstrap does not claim to
+      # require a manual user intervention.
+      if [ "${NONINTERACTIVE}" = 1 -a "${INTERACTIVE_REBOOTSTRAP}" != 1 ]; then
```

Comment:
We've gotten inconsistent with regard to spelling: "$VAR" vs. "${VAR}". I lean toward the former, for conciseness, unless we're trying to concatenate a string with it, as `${foo}bar`.

Reply:
OK I do that. I prefer version `${}` because it is more explicit, and I saw the advice to always use it several times. But I do not intend to change every variable in `certbot-auto`, and global readability is even more important than some linting rules.